packages/nimble-components/src/anchor/specs/README.md

```diff
@@ -0,0 +1,203 @@
+-   [RouterLinkWithHref](https://github.com/angular/angular/blob/0a2191f8e7e232087aab0a7a9eb9ee6871580267/packages/router/src/directives/router_link.ts#L257): Selector `a[routerLink],area[routerLink]`: Computes an href for the anchor element it targets; left-click does a router navigation; Ctrl-click and middle-mouse-button click defer to the browser (to open a new tab/window)
+
+We want the behavior of RouterLinkWithHref, so we can subclass it and update the selector to target `nimble-anchor`.
+However, if we have clients use the existing `routerLink` attribute to opt into our directive, there'll still be a RouterLink directive active doing the wrong action too. In current Angular versions there's not a good way to disable RouterLink navigation ([see the comment on this Angular commit](https://github.com/angular/angular/commit/ccb09b4558a3864fb5b2fe2214d08f1c1fe2758f)).
+
+We will follow the same approach previously used for the BreadcrumbItem:
+
+-   Our directive's selector will be `nimble-anchor[nimbleRouterLink]`, and the directive will define `@Input nimbleRouterLink` (which sets `routerLink`). This is a small change for clients which we will document (consistent with BreadcrumbItem), and other routerLink attributes can still be used as-is:
+    ```html
+    <nimble-anchor
+        nimbleRouterLink="/customapp"
+        [queryParams]="{debug: true}"
+        [state]="{tracingId: 123}"
+    >
+        Custom App Page
+    </nimble-anchor>
+    ```
+-   A second directive will be provided to disable use of the `routerLink` attribute on `nimble-anchor` elements.
+
+Another option would be to try to reuse the two directives created for the BreadcrumbLink for these purposes (i.e. NimbleBreadcrumbItemRouterLinkWithHrefDirective and NimbleBreadcrumbItemRouterLinkDirective). There is so little in these directives, though, that it is simpler just to create duplicates for the anchor.
+
+### Blazor integration
+
+We will create Blazor wrappers for the anchor and button-like anchor. Judging by the BreadcrumbItem's wrapper, there are no special routing considerations/mechanisms for Blazor.
+
+### Additional requirements
+
+-   _User interaction:_ None
+-   _Styling:_
+    -   When it has keyboard focus, the anchor will have a double underline.
+    -   CSS for button styles will be shared as much as possible
+    -   CSS for hyperlink styles will be shared as much as possible with the Breadcrumb
+-   _Testing:_ None
+-   _Documentation:_ Should direct users to set `inline` when using the anchor inline with text.
+-   _Tooling:_ None
+-   _Accessibility:_ Need to `applyMixins` on the Nimble anchor type, from the `DelegatesARIALink` class.
```

Comment:
For my education, could you add more detail about the anchor button's accessibility? For example, will it have `role=button` or `role=link`?

Reply:
I added a couple sentences. Basically, it's a `link` because it works like a link and the shadow root contains a `<a>` element that naturally has the role `link`.